k-distribution/tutorial/1_k/1_lambda/lesson_9/lambda.md

```diff
@@ -0,0 +1,145 @@
+**K** concepts and how they work in the K tool, and not to teach
+λ-calculus or to argue for one definitional style against another
+(e.g., some may prefer environment/closure-based definitions of such
+languages).
+
+Note that the subsequent definition is so simple, that it hardly shows any
+of the strengths of **K**.  Perhaps the most interesting **K** aspect it shows is
+that substitution can be defined fully generically, and then used to give
+semantics to various constructs in various languages.
+
+Note:
+ **K** follows the
+[literate programming](https://en.wikipedia.org/wiki/Literate_programming)
+approach. The various semantic features defined in a **K**
+module can be reordered at will and can be commented using normal
+comments like in C/C++/Java.
+While comments are useful in general, they can annoy the expert user
+of **K**. To turn them off, you can do one of the following (unless you
+want to remove them manually):  
+(1) Use an [editor](https://github.com/kframework/k-editor-support) which can
+hide or highlight Markdown and conventional C-like comments; or  
+(2) Run `kompile --debug <def>`. Inside `./.kompiled-xxx/.md2.k/` you will find
+all the K code extracted from the markdown files as used for compilation.
+
+### Substitution
+We need the predefined substitution module, so we require it with the command
+below.  Then we should make sure we import its module called SUBSTITUTION
+in our LAMBDA module below.
+
+```k
+require "substitution.md"
+
+module LAMBDA
+  imports DOMAINS
+  imports SUBSTITUTION
+```
+### Basic Call-by-value λ-Calculus
+
+We first define a conventional call-by-value λ-calculus, making sure
+we declare the lambda abstraction construct to be a binder, the
+lambda application to be strict, and the parentheses used for grouping as
+a bracket.
+
+Note:
+Syntax in **K** is defined using the familiar BNF notation, with
+terminals enclosed in quotes and nonterminals starting with capital
+letters. **K** actually extends BNF with several attributes, which will be
+described in this tutorial.
+
+Note:
+The `strict` constructs can evaluate their arguments in any (fully
+interleaved) order.
+
+
+The initial syntax of our λ-calculus:
+```k
+  syntax Val ::= KVar
+               | "lambda" KVar "." Exp  [binder, latex(\lambda{#1}.{#2})]
+  syntax Exp ::= Val
+               | Exp Exp              [left, strict]
+               | "(" Exp ")"          [bracket]
+  syntax KResult ::= Val
+```
+### β-reduction
+
+```k
+  rule (lambda X:KVar . E:Exp) V:Val => E[V / X]
+```
+
+### Integer and Boolean Builtins
+The LAMBDA arithmetic and Boolean expression constructs are simply rewritten
+to their builtin counterparts once their arguments are evaluated.
+The operations with subscripts in the right-hand sides of the rules below are
+builtin and come with the corresponding builtin sort; they are actually
+written like `+Int` in ASCII, but they have LaTeX attributes to be
```

Comment:
this is another reference that should probably be adjusted

Reply:
This is referring to the latex attributes.
When I converted imp for the first time, you (or Everett) asked me not to remove them.
So what do we want to do with them? At this moment I don't see how they are useful.